code/game/objects/items/weapons/tanks/tanks.dm

```diff
@@ -141,14 +141,24 @@
 	ui_interact(user)
 
 /obj/item/weapon/tank/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	// update the ui if it exists, returns null if no ui is passed/found
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		// the ui does not exist, so we'll create a new() one
+        // for a list of parameters and their descriptions see the code docs in \code\modules\nano\nanoui.dm
+		ui = new(user, src, ui_key, "tanks.tmpl", "Tank", 500, 300)
+		// open the new ui window
+		ui.open()
+		// auto update every Master Controller tick
+		ui.set_auto_update(1)
 
+/obj/item/weapon/tank/ui_data(mob/user, datum/topic_state/state = default_state)
 	var/using_internal
-	if(istype(loc,/mob/living/carbon))
-		var/mob/living/carbon/location = loc
-		if(location.internal==src)
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
+		if(C.internal == src)

```

Comment:
another minor change in the middle of a big code shift that is also still 100% ok by me